src/huggingface_hub/hf_file_system.py

```diff
@@ -375,34 +376,35 @@ def info(self, path: str, **kwargs) -> Dict[str, Any]:
             return {"name": name, "size": 0, "type": "directory"}
         return super().info(path, **kwargs)
 
-    def expand_path(
-        self, path: Union[str, List[str]], recursive: bool = False, maxdepth: Optional[int] = None, **kwargs
-    ) -> List[str]:
-        # The default implementation does not allow passing custom kwargs (e.g., we use these kwargs to propagate the `revision`)
-        if maxdepth is not None and maxdepth < 1:
-            raise ValueError("maxdepth must be at least 1")
-
-        if isinstance(path, str):
-            return self.expand_path([path], recursive, maxdepth)
-
-        out = set()
-        path = [self._strip_protocol(p) for p in path]
-        for p in path:
-            if has_magic(p):
-                bit = set(self.glob(p, **kwargs))
-                out |= bit
-                if recursive:
-                    out |= set(self.expand_path(list(bit), recursive=recursive, maxdepth=maxdepth, **kwargs))
-                continue
-            elif recursive:
-                rec = set(self.find(p, maxdepth=maxdepth, withdirs=True, detail=False, **kwargs))
-                out |= rec
-            if p not in out and (recursive is False or self.exists(p)):
-                # should only check once, for the root
-                out.add(p)
-        if not out:
-            raise FileNotFoundError(path)
-        return list(sorted(out))
+    if version.parse(fsspec.__version__) < version.parse("2023.5.0"):
+        # The default implementation in fsspec<2023.5.0 does not allow passing custom kwargs (e.g., we use these kwargs to propagate the `revision`)
+        def expand_path(
+            self, path: Union[str, List[str]], recursive: bool = False, maxdepth: Optional[int] = None, **kwargs
+        ) -> List[str]:
```

Comment:
What happens then on recent versions (`fsspec>=2023.5.0`)? It defaults back to the `expand_path` implemented in fsspec directly?

Reply:
Yes, `expand_path` allows passing `**kwargs` in versions >=2023.5.0 (thanks to https://github.com/fsspec/filesystem_spec/commit/834115f2b91f61d09004beb1bfd46d705a201c2b), so there is no need to override it.

Reply:
Ah I see, so the implementation is just a copy of the official fsspec one.
In your opinion, could we pin `fsspec>=2023.5.0` in our dependencies to avoid having to support older versions? 
Otherwise if you feel strongly against it, PR looks good to me. I'd just add a comment linking to https://github.com/fsspec/filesystem_spec/blob/4f0eb488f26299592df5cc6494a5ebd8cb8cca4f/fsspec/spec.py#L1139 to say that it's a duplicate (at first I understood that we has a custom implementation here, in addition of the kwargs).

Reply:
Bumping the minimal version sounds good to me :)

Reply:
For instance, Colab builds the env with version `2023.6.0`, so this should have little impact on the user experience.

Reply:
Good to know! Thanks for making the change.